### ISSUE TITLE
Issues about local attention in PVP module

### ISSUE THREAD

**Issue by Iriswwhy** [zhangzw12319/lcps#14]
Thank you very much for the excellent work.
I found the local attention calculation seems to be different in the text part of Section 3.4 in the paper, Figure 3.(b) and the code. In section 3.4, you used fused point features as neighboring voxels and they represent keys and values. In Figure 3.(b), the original voxels act as neighboring voxels and they represent Query and Value. In the code in cylinder_fea_generator.py, the code is:

`if self.use_att:
                CHANNEL = fusion_pooled.shape[1]
                shifted_index = shift_voxel_grids(unq_no_zero, INDEX_SHIFT, self.grid_size, cur_dev) # (27, sp_v_n, 3)
                for i, each_shift_index in enumerate(shifted_index):
                    # select_ind = [return_tensor_index(value=shift_ind, t=unq_no_zero) for shift_ind in each_shift_index] # too slow
                    select_ind = return_tensor_index_v2(value=each_shift_index, t=unq_no_zero) # (27, 2)
                    select_ind = torch.tensor(select_ind, device=cur_dev) # (sp_v_n,)
                    condition = (select_ind >= 0).unsqueeze(1).expand_as(fusion_pooled) # (sp_v_n, CHANNEL)
                    tmp = fusion_pooled[select_ind] # (sp_v_n, CHANNEL)
                    tmp= tmp.masked_fill(~condition, 0) # fill zero if select_ind  == -1
                    key_value_list.append(tmp)     
                key = torch.stack(key_value_list)
                key = key.permute(1, 0, 2) # (sp_v_n, 27, CHANNEL)
                feat_query = fusion_pooled.unsqueeze(1) # (sp_v_n, 1, CHANNEL)
                out, _att_weights = self.atten_fusion(feat_query, key, key) # (sp_v_n, 1, CHANNEL)
                out = out.squeeze(1)
                cat_att_fea = torch.cat((fused_cylinder_data, out), dim=0)
                fused_cylinder_data = torch_scatter.scatter_add(cat_att_fea, unq_inv_1, dim=0)`

It seems that the key and feat_query are both from fused point features. And the cat_att_fea ( = torch.cat((fused_cylinder_data, out), dim=0)) seems to be different from the equation [θ = Att(θp, θim, θim) + θp]? Does **out** refer to Att(θp, θim, θim) and **fused_cylinder_data** refer to θp? θp is the original cylindrical voxels denoted in the paper. 
It is a little confusing, I wonder which one is correct. Thanks.
